### PR TITLE
fix(toaster): Do not call `ReactDOM` render multiple times

### PR DESCRIPTION
**Pull request by veekeys**
I have changed approach a little in toaster not to call `render` multiple times (as this is not the recommended way to do ).
I changed that mostly because of my v18 work, where tests with new root render were not passing for some reason.
With this change toasts will use react change detection and rerender as state of the component will change.
Using fancy [useImperativeHandle](https://reactjs.org/docs/hooks-reference.html#useimperativehandle) hook.

### CODE REVIEW

File: packages/iTwinUI-react/src/core/Toast/Toaster.tsx [hunk 1]
```diff
@@ -55,6 +65,7 @@ export default class Toaster {
       ? 'ascending'
       : 'descending';
     this.settings = newSettings;
+    this.toastsRef?.current?.setPlacement(this.settings.placement ?? 'top');
```
Comment (bentleyvk) [inline packages/iTwinUI-react/src/core/Toast/Toaster.tsx:68]: DO you need first `?`?

File: packages/iTwinUI-react/src/core/Toast/Toaster.test.tsx [hunk 2]
```diff
@@ -2,6 +2,7 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
+import { act } from 'react-dom/test-utils';
```
Comment (bentleyvk) [inline packages/iTwinUI-react/src/core/Toast/Toaster.test.tsx:5]: Why imported not from testing-library?

File: packages/iTwinUI-react/src/core/Toast/ToastWrapper.test.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@
 import React from 'react';
 import { render } from '@testing-library/react';
 
-import { ToastWrapper } from './ToastWrapper';
+import { ToastWrapper, ToastWrapperHandle } from './ToastWrapper';
 import { ToastProps } from './Toast';
 import { act } from 'react-dom/test-utils';
```
Comment (bentleyvk) [inline packages/iTwinUI-react/src/core/Toast/ToastWrapper.test.tsx:10]: Why act is imported from here???

File: packages/iTwinUI-react/src/core/Toast/Toaster.tsx (hunk 1 above)
Comment (veekeys) [reply packages/iTwinUI-react/src/core/Toast/Toaster.tsx]: no

File: packages/iTwinUI-react/src/core/Toast/Toaster.test.tsx (hunk 2 above)
Comment (veekeys) [reply packages/iTwinUI-react/src/core/Toast/Toaster.test.tsx]: I'm just gonna quote this
https://github.com/iTwin/iTwinUI-react/pull/623#discussion_r871357029

But removed all use cases of that import..

File: packages/iTwinUI-react/src/core/Toast/ToastWrapper.tsx [hunk 4]
```diff
@@ -8,14 +8,31 @@ import cx from 'classnames';
...
+};
+
+export const ToastWrapper = React.forwardRef<ToastWrapperHandle>((_, ref) => {
+  const [toasts, setToasts] = React.useState<ToastProps[]>([]);
+  const [placement, setPlacement] = React.useState<ToastPlacement>('top');
   const placementPosition = placement.startsWith('top') ? 'top' : 'bottom';
 
+  React.useImperativeHandle(
+    ref,
+    () => ({
```
[the remark this answers is not in the document]
Comment (veekeys) [reply packages/iTwinUI-react/src/core/Toast/ToastWrapper.tsx:25]: changed

File: packages/iTwinUI-react/src/core/Toast/ToastWrapper.test.tsx (hunk 3 above)
Comment (veekeys) [reply packages/iTwinUI-react/src/core/Toast/ToastWrapper.test.tsx]: But removed all use cases of that import..

**Review by adamhock — APPROVED** [review]
Looks good to me :tada:

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**